sympy/solvers/polysys.py

```diff
@@ -83,9 +80,15 @@ def solve_biquadratic(f, g, opt):
     x, y = opt.gens
 
     p = Poly(p, x, expand=False)
-    q = q.ltrim(-1)
-
     p_roots = [ rcollect(expr, y) for expr in roots(p).keys() ]
+    if len(p_roots) == 1 and y not in p_roots[0].free_symbols:
+        # this is not a zero-dimensional system
```

Comment:
It seems that this test is not reliable. Consider `(x**2 + y**2 - 1, y**2 - 1)`. I'd detect systems of non-zero dimension by checking if `gcd(f, g)` is constant or not.